mbedtls/build.rs

```diff
@@ -11,17 +11,36 @@ use std::env;
 
 use rustc_version::Channel;
 
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+  
 /// Return the crate hash that Cargo will be passing to `rustc -C metadata=`.
 // If there's a panic in this code block, that means Cargo's way of running the
 // build script has changed, and this code should be updated to handle the new
 // case.
-fn get_compilation_metadata_hash() -> String {
+// Since Bazel is not follow the Cargo OUT_DIR structure and does not generate a hash to provide to rustc, we need to implement it in build script it self.
+fn get_compilation_symbol_suffix() -> String {
     let out_dir: std::path::PathBuf = std::env::var_os("OUT_DIR").unwrap().into();
     let mut out_dir_it = out_dir.iter().rev();
-    assert_eq!(out_dir_it.next().unwrap(), "out");
-    let crate_ = out_dir_it.next().unwrap().to_string_lossy();
-    assert!(crate_.starts_with("mbedtls-"));
-    crate_[8..].to_owned()
+    let next = out_dir_it.next().unwrap();
+
+    if next == "out" {
+        // If Cargo is used as build system.
+        let crate_ = out_dir_it.next().unwrap().to_string_lossy();
+        assert!(crate_.starts_with("mbedtls-"));
+        return crate_[8..].to_owned();
+    } else if next == "_bs.out_dir" {
+        // If Bazel is the build system.
+        let version = env!("CARGO_PKG_VERSION");
+        let mut hasher = DefaultHasher::new();
+        let context = format!("{}{}", "mbedtls", version);
```

Comment:
I suppose we could use the `CARGO_PKG_NAME` env var instead of hardcoding `mbedtls`?

Also, since we're already already hashing different components together, there's no need to use `format!()` to combine the name and version, the hasher already does that.